lib/src/player.dart

```diff
@@ -123,17 +123,28 @@ class Player with PlayerEventHandler implements PlayerApi {
 
   // Can be used to call methods on the platform side that return a single
   // primitive value that is natively supported by the method channel.
-  Future<T?> _invokeMethod<T>(
+  Future<T?> _invokeMethodNullable<T>(
     String methodName, [
     dynamic data,
   ]) async {
     final result = await _initializationResult;
     if (!result) {
-      return Future.error('Error initializing player on native platform side.');
+      throw Exception('Error initializing player on native platform side.');
     }
     return _methodChannel.invokeMethod<T>(methodName, _buildPayload(data));
   }
 
+  Future<T> _invokeMethod<T>(
+    String methodName, [
+    dynamic data,
+  ]) async {
+    final result = await _invokeMethodNullable<T>(methodName, data);
+    if (result == null) {
+      throw Exception('Invalid type returned from the native platform side.');
```

Comment:
Let's also print the method name for better debugging:
```suggestion
      throw Exception('Invalid type returned for $methodName from the native platform side.');
```

Reply:
Done

Reply:
Done